frontend/src/utils/validationSchema.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from "zod";
+
+export const signUpSchema = z.object({
+  name: z.string().min(5, "Name must be at least 5 characters"),
+  email: z.string().email("Invalid email address").min(1, "Email is required"),
+  password: z.string().min(6, "Password must be at least 6 characters"),
+});
+
```

Comment:
you can store the validation messages inside `constants/` directory as well. Make sure not to repeate things as far as possible.